subsys/bluetooth/host/conn.c

```diff
@@ -115,10 +115,12 @@ struct k_sem *bt_conn_get_pkts(struct bt_conn *conn)
 	}
 #endif /* CONFIG_BT_BREDR */
 #if defined(CONFIG_BT_ISO)
-	if (conn->type == BT_CONN_TYPE_ISO || bt_dev.le.iso_mtu) {
-		if (bt_dev.le.iso_pkts.limit) {
-			return &bt_dev.le.iso_pkts;
-		}
+	/* Use ISO pkts semaphore if LE Read Buffer Size command returned
+	 * dedicated ISO buffers.
+	 */
+	if (conn->type == BT_CONN_TYPE_ISO &&
+	    bt_dev.le.iso_mtu && bt_dev.le.iso_pkts.limit) {
+		return &bt_dev.le.iso_pkts;
```

Comment:
This seems more correct, and more similar to the check we have in `conn_mtu`:
```c
static inline uint16_t conn_mtu(struct bt_conn *conn)
{
#if defined(CONFIG_BT_BREDR)
	if (conn->type == BT_CONN_TYPE_BR || !bt_dev.le.acl_mtu) {
		return bt_dev.br.mtu;
	}
#endif /* CONFIG_BT_BREDR */
#if defined(CONFIG_BT_ISO)
	if (conn->type == BT_CONN_TYPE_ISO && bt_dev.le.iso_mtu) {
		return bt_dev.le.iso_mtu;
	}
#endif /* CONFIG_BT_ISO */
#if defined(CONFIG_BT_CONN)
	return bt_dev.le.acl_mtu;
#else
	return 0;
#endif /* CONFIG_BT_CONN */
}
```

where it is an `&&` instead of an `||`. 

I do wonder if it should be something like
```c
	if (conn->type == BT_CONN_TYPE_ISO) {
		if (bt_dev.le.iso_mtu && bt_dev.le.iso_pkts.limit) {
			return &bt_dev.le.iso_pkts;
		}

		return NULL;
	}
```

So that we return `0` for ISO connections if we haven't read the buffer count yet, or if the buffer count is 0? 

Maybe it could even be as simple as 
```c
	if (conn->type == BT_CONN_TYPE_ISO) {
		return &bt_dev.le.iso_pkts;
	}
```

The point is that we probably don't want to return `&bt_dev.le.acl_pkts;` for ISO channels at all, right? 

Should `conn_mtu` be updated too?

Reply:
`If the Controller supports HCI ISO
Data packets, it shall return non-zero values for the ISO_Data_Packet_Length
and Total_Num_ISO_Data_Packets parameters.`

By this statement, we should use your proposal, right?:
```c
	if (conn->type == BT_CONN_TYPE_ISO) {
		if (bt_dev.le.iso_mtu && bt_dev.le.iso_pkts.limit) {
			return &bt_dev.le.iso_pkts;
		}

		return NULL;
	}
```

Reply:
> `If the Controller supports HCI ISO Data packets, it shall return non-zero values for the ISO_Data_Packet_Length and Total_Num_ISO_Data_Packets parameters.`
> 
> By this statement, we should use your proposal, right?:
> 
> ```c
> 	if (conn->type == BT_CONN_TYPE_ISO) {
> 		if (bt_dev.le.iso_mtu && bt_dev.le.iso_pkts.limit) {
> 			return &bt_dev.le.iso_pkts;
> 		}
> 
> 		return NULL;
> 	}
> ```

Yes, I think this would be the best way of handling it, otherwise we risk returning a non-NULL value for ISO packets (which would for connected ISO default to the ACL buffers).